local-volume/provisioner/pkg/common/common.go

```diff
@@ -167,28 +158,61 @@ func GetDefaultVolumeConfig() map[string]MountConfig {
 }
 
 // VolumeConfigToConfigMapData converts volume config to configmap data.
-func VolumeConfigToConfigMapData(config map[string]MountConfig) (map[string]string, error) {
+func VolumeConfigToConfigMapData(config *ProvisionerConfiguration) (map[string]string, error) {
 	configMapData := make(map[string]string)
-	for class, data := range config {
-		var val []byte
-		var err error
-		if val, err = json.Marshal(data); err != nil {
-			return nil, fmt.Errorf("unable to unmarshal config for class %v: %v", class, err)
-		}
-		configMapData[class] = string(val)
+	val, err := yaml.Marshal(config.StorageClassConfig)
+	if err != nil {
+		return nil, fmt.Errorf("unable to Marshal volume config: %v", err)
 	}
+	configMapData[ProvisonerStorageClassConfig] = string(val)
 	return configMapData, nil
 }
 
 // ConfigMapDataToVolumeConfig converts configmap data to volume config
-func ConfigMapDataToVolumeConfig(data map[string]string) (map[string]MountConfig, error) {
-	mountConfig := make(map[string]MountConfig)
-	for class, val := range data {
-		config := MountConfig{}
-		if err := json.Unmarshal([]byte(val), &config); err != nil {
-			return nil, fmt.Errorf("unable to unmarshal config for class %v: %v", class, err)
+func ConfigMapDataToVolumeConfig(data map[string]string, provisionerConfig *ProvisionerConfiguration) error {
+	rawYaml := ""
+	for key, val := range data {
+		rawYaml += key
+		rawYaml += ": \n"
+		rawYaml += insertSpaces(string(val))
+	}
+	if err := yaml.Unmarshal([]byte(rawYaml), provisionerConfig); err != nil {
+		return fmt.Errorf("fail to Unmarshal yaml due to: %#v", err)
+	}
+	return nil
+}
+
+func insertSpaces(original string) string {
+	spaced := ""
+	for _, line := range strings.Split(original, "\n") {
+		spaced += "   "
+		spaced += line
+		spaced += "\n"
+	}
+	return spaced
+}
+
+// LoadProvisionerConfigs loads all configuration into a string and unmarshal it into ProvisionerConfiguration struct.
+func LoadProvisionerConfigs(provisionerConfig ProvisionerConfiguration) error {
```

Comment:
Can you add a comment that this is for parsing the config map mounted as a volume.